src/xrpld/rpc/handlers/LedgerEntry.cpp

```diff
@@ -66,768 +67,819 @@ parseAuthorizeCredentials(Json::Value const& jv)
     return arr;
 }
 
-// {
-//   ledger_hash : <ledger>
-//   ledger_index : <ledger_index>
-//   ...
-// }
-Json::Value
-doLedgerEntry(RPC::JsonContext& context)
+std::optional<uint256>
+parseIndex(Json::Value const& params, Json::Value& jvResult)
 {
-    std::shared_ptr<ReadView const> lpLedger;
-    auto jvResult = RPC::lookupLedger(lpLedger, context);
+    uint256 uNodeIndex;
+    if (!uNodeIndex.parseHex(params.asString()))
+    {
+        jvResult[jss::error] = "malformedRequest";
+        return std::nullopt;
+    }
+    return uNodeIndex;
+}
 
-    if (!lpLedger)
-        return jvResult;
+std::optional<uint256>
+parseAccountRoot(Json::Value const& params, Json::Value& jvResult)
+{
+    auto const account = parseBase58<AccountID>(params.asString());
+    if (!account || account->isZero())
+    {
+        jvResult[jss::error] = "malformedAddress";
+        return std::nullopt;
+    }
+    return keylet::account(*account).key;
+}
 
+std::optional<uint256>
+parseCheck(Json::Value const& params, Json::Value& jvResult)
+{
     uint256 uNodeIndex;
-    bool bNodeBinary = false;
-    LedgerEntryType expectedType = ltANY;
+    if (!uNodeIndex.parseHex(params.asString()))
+    {
+        jvResult[jss::error] = "malformedRequest";
+        return std::nullopt;
+    }
+    return uNodeIndex;
+}
 
-    try
+std::optional<uint256>
+parseDepositPreauth(Json::Value const& dp, Json::Value& jvResult)
+{
+    if (!dp.isObject())
     {
-        if (context.params.isMember(jss::index))
+        uint256 uNodeIndex;
+        if (!dp.isString() || !uNodeIndex.parseHex(dp.asString()))
         {
-            if (!uNodeIndex.parseHex(context.params[jss::index].asString()))
-            {
-                uNodeIndex = beast::zero;
-                jvResult[jss::error] = "malformedRequest";
-            }
+            jvResult[jss::error] = "malformedRequest";
+            return std::nullopt;
         }
-        else if (context.params.isMember(jss::account_root))
+        return uNodeIndex;
+    }
+    // clang-format off
+    if (
```

Comment:
This can be improved by splitting to some constants and/or helper function(s) 👍 Can be done in a later PR though.